components-core/src/main/java/com/adyen/checkout/components/core/internal/analytics/AnalyticsManager.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2024 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by oscars on 8/3/2024.
+ */
+
+package com.adyen.checkout.components.core.internal.analytics
+
+import androidx.annotation.RestrictTo
+import kotlinx.coroutines.CoroutineScope
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+interface AnalyticsManager {
+
+    fun initialize(owner: Any, coroutineScope: CoroutineScope)
```

Comment:
instead of having the owner to be the type of `Any` we can introduce an interface like `AnalyticsOwner` that would be implemented in `DropInViewModel` and `delegates` so we would be more precise what we need here. that could also prevent undesired calls for this function (not sure if that would ever happen, but if we try to call this in a place we would immediately know it should be an `AnalyticsOwner`

Reply:
Fair point. I'm thinking any class that implements that interface gets to define owner the way it wants to. I think we should avoid that somehow. I can't think of a way other than an interface for that. Also not sure if I'm being overly careful with this. @araratthehero I'm curious what you think about this?

Reply:
Having an interface for this case will add extra work every time you want to use analytics manager. Analytics usage should be as simple as possible without extra implementations. 
Since we need the `owner` only for identification purposes, it is best to not introduce a limitation for the owner to be of a specific type and keep it as simple as possible.
`owner::class.qualifiedName` is generic and gives a unique value, which solves the issue for our use case, to identify if `clear()` function is called from the same owner. If we add extra usage for the owner other than identification, I would suggest reviewing what we have.

But if you have any idea how we can identify the caller without extra implementations or getting the `qualifiedName` of the caller class would be good to discuss.